src/controllers/guildRoleHandler.ts

```diff
@@ -68,3 +70,74 @@ export async function removeGuildRoleHandler(request: IRequest, env: env) {
     });
   }
 }
+export async function getGuildRolesHandler(request: IRequest, env: env) {
+  const authHeader = request.headers.get("Authorization");
+
+  if (!authHeader) {
+    return new JSONResponse(response.BAD_SIGNATURE, { status: 401 });
+  }
+  try {
+    await verifyAuthToken(authHeader, env);
+    const roles = await getGuildRoles(env);
+    return new JSONResponse({ roles });
+  } catch (err: any) {
+    const error =
+      err?.message === response.ROLE_FETCH_FAILED
+        ? response.ROLE_FETCH_FAILED
+        : response.INTERNAL_SERVER_ERROR;
+    return new JSONResponse(
+      {
+        error,
+      },
+      {
+        status: 500,
+        headers: {
+          "content-type": "application/json;charset=UTF-8",
+        },
+      }
+    );
+  }
+}
+
+export async function getGuildRoleByRoleNameHandler(
+  request: IRequest,
+  env: env
+) {
+  const authHeader = request.headers.get("Authorization");
+  const roleName = decodeURI(request.params?.roleName ?? "");
+
+  if (!authHeader) {
+    return new JSONResponse(response.BAD_SIGNATURE, { status: 401 });
+  }
+
+  if (!roleName) {
+    return new JSONResponse(response.NOT_FOUND, { status: 404 });
+  }
+  try {
+    await verifyAuthToken(authHeader, env);
+    const role = await getGuildRoleByName(roleName, env);
+    if (!role) {
+      return new JSONResponse(response.NOT_FOUND, {
+        status: 404,
+        headers: {
+          "content-type": "application/json;charset=UTF-8",
+        },
+      });
+    }
+    return new JSONResponse(role);
+  } catch (err: any) {
+    const error =
+      err?.message === response.ROLE_FETCH_FAILED
```

Comment:
please change it over here also

Reply:
Thank you for your comment. But I did try that. 
![image](https://github.com/Real-Dev-Squad/discord-slash-commands/assets/27444637/2fbdf3f7-dbb5-49c4-a75f-894f891d408d)

But I get the above error. 
Let me know if you think this is a important change for you?